django_dbq/models.py

```diff
@@ -58,7 +58,9 @@ class Meta:
     objects = JobManager()
 
     def save(self, *args, **kwargs):
-        if not self.pk:
+        is_new = not Job.objects.filter(pk=self.pk).exists()
```

Comment:
Tried that. The tests fail. They fail because when the worker is running in a real app, the state is prodded externally by the worker process as the jobs are processed. In the tests, this prodding wasn't happening. I could do the prodding in the tests, but it feels a bit fragile..